docs/gateway/02_api_spec.md

```diff
@@ -0,0 +1,72 @@
+# API Specification
+
+## Base URLs
+
+All end-points here are specified with reference to these base URLs:
+
+| Environment       | URL                           | Usage                                  |
+|-------------------|-------------------------------|----------------------------------------|
+| Local development | `http://localhost:5000/`      | Local product development              | 
+| QA                | `https://qa-lms.hypothes.is/` | Publicly available testing environment |
+| Production        | `https://lms.hypothes.is/`    | Final deployment with real users       |
+
+Unless you are familiar with Hypothesis development, we recommend you start 
+with the public QA test environment.
+
+## POST `/lti_launches`*
+
+_* Note: This is the main HTML launch URL, a new URL will be provided for
+the gateway soon._
+
+This end-point will allow you to test the authentication mechanism while the 
+final end-point is being written.
+
+### Headers
+```
+Accept: text/html
+Content-Type: application/x-www-form-urlencoded
+```
+
+_Note: The final end-point will return `application/json` not HTML_
+
+### Post body fields
+
+We accept any and all LTI 1.1 parameters.
+
+ * See "Basic Launch Data" in https://www.imsglobal.org/specs/ltiv1p1p1/implementation-guide
+
+It is recommended for stability and the best performance that you pass all LTI
+parameters received onto this end-point. This will ensure compatibility as the
+more features may be added in future which require fields which were previously
+optional.
+
+| Field                         | Example                    | Notes                                                                   |
+|-------------------------------|----------------------------|-------------------------------------------------------------------------|
+| `lti_version`                 | `LTI-1p0`                  | Specify LTI version (only 1.1 supported at present)                     |
+| `lti_message_type`            | `basic-lti-launch-request` | Indicate you are making a launch                                        |
+| `tool_consumer_instance_guid` | `ab5e8623477ffe8324`       | The unique ID for the tool consumer instance (e.g. Canvas installation) |
+| `user_id`                     | `2978763`                  | The user id we should act on behalf of                                  |
+| `roles`                       | `Instructor`               | The role that user has (used in permission generation)                  |
+| `context_id`                  | `454`                      | Specify the course you want to access                                   |
+| `context_title`               | `My course name`           | _Note: This is only a required field for now_                           |
```

Comment:
Is  GUID not required as well?

Reply:
This is mentioned above. What I'm trying to say here is that `context_title` won't be a required field in future, rather than it's the only one.